californium-core/src/main/java/org/eclipse/californium/core/observe/Observation.java

```diff
@@ -52,6 +52,20 @@ public Request getRequest() {
 	}
 
 	/**
+	 * @return a cloned observation with a shallow clone of request
+	 */
+	public Observation cloneObservation() {
```

Comment:
However, for the same reason I do not think that it is a good idea to move the *cloning* code into `Observation`. The fact that we need to clone the `Observation` object by doing this *shallow* copying is actually a property of the `InMemoryObservationStore`. A DB based store would not need to do this because when reading from the DB a new object would be instantiated ...